### PR TITLE
Fix Module.is_public() when the module is not a the root

### PR DESCRIPTION
**Pull request by tibdex**
Thanks for submitting a PR!

Please make sure to check for the following items:
- [x] Add unit tests and integration tests where applicable.  
      If you've added an error code or changed an error code behavior,
      you should probably add or change a test case file under `tests/test_cases/` and add 
      it to the list under `tests/test_definitions.py`.  
      If you've added or changed a command line option,
      you should probably add or change a test in `tests/test_integration.py`.
- [x] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".  
      Make sure to include the PR number after you open and get one.
   
Please don't get discouraged as it may take a while to get a review.

### CODE REVIEW

File: src/pydocstyle/parser.py [hunk 1]
```diff
@@ -117,7 +118,8 @@ def is_public(self):
 
         This helps determine if it requires a docstring.
         """
-        return not self.name.startswith('_') or self.name.startswith('__')
+        module_name = os.path.basename(self.name)
```
Comment (Nurdok) [inline src/pydocstyle/parser.py:121]: Can you please use `pathlib` instead of `os.path`?

**Comment by samj1912** [comment]
@tibdex I think this is a good patch, can you please switch to pathlib so that we can merge this :)

File: src/tests/parser_test.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 """Parser tests."""
 
 import io
+import os
```
Comment (Nurdok) [inline src/tests/parser_test.py:4]: Please use `pathlib` in this file, too.

**Comment by samj1912** [comment]
@tibdex bump?

File: src/tests/parser_test.py (hunk 2 above)
Comment (gbroques) [reply src/tests/parser_test.py:4]: Updated in PR #494.

**Comment by tibdex** [comment]
See https://github.com/PyCQA/pydocstyle/pull/494#issuecomment-664320247.

**Pull request closed without merging**